subsys/bluetooth/host/hci_core.c

```diff
@@ -943,11 +943,10 @@ static void le_enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 	}
 
 	if (BT_FEAT_LE_DLE(bt_dev.le.features)) {
-		err = hci_le_set_data_len(conn);
-		if (!err) {
-			atomic_set_bit(conn->flags, BT_CONN_AUTO_DATA_LEN);
-			goto done;
-		}
+		/* data len changed event is optional so we just ignore error
+		 * here
+		 */
+		hci_le_set_data_len(conn);
```

Comment:
Shouldn't you just change the return type of this function to void? You can then have a comment in the function definition yourself, and then you don't need to provide this excuse every time you call it. Bonus: this way we also don't violate the recent MISRA-C updates which require void casting whenever you ignore the return value of a non-void function.

Reply:
One other thing: when you do this change, add appropriate BT_ERR() logs to the error return branches so that such failures don't go completely unnoticed.